lib/data/shared/service/remote/network.dart

```diff
@@ -24,6 +26,8 @@ abstract class Network {
           // AuthBasicInterceptor(base64Credentials),
           // Add API Key
           ApiKeyInterceptor(apiKey),
+          // Firebase Performance Monitoring
+          if (!debugMode) FirebasePerformanceInterceptor(),
```

Comment:
IIRC there is already a global variable `kDebugMode` to check if the app is running in debug mode so we don't need to pass it as a function parameter. I think we already use it when calling `Dependencies.register()` so maybe it's a bit redundant to pass the param through several functions when we can directly access the var.